runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -114,6 +136,17 @@ initImpl(J9VMThread *currentThread, j9object_t membernameObject, j9object_t refO
 
 		J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(methodID->method);
 
+		if (isPolymorphicMHMethod(currentThread->javaVM, J9_CLASS_FROM_METHOD(methodID->method), J9ROMMETHOD_NAME(romMethod))
+#if JAVA_SPEC_VERSION >= 9
+		|| ((J9_CLASS_FROM_METHOD(methodID->method) == J9VMJAVALANGINVOKEVARHANDLE(currentThread->javaVM))
+			&& VM_VMHelpers::isPolymorphicVarHandleMethod(J9UTF8_DATA(J9ROMMETHOD_NAME(romMethod)),J9UTF8_LENGTH(J9ROMMETHOD_NAME(romMethod)))
+			)
+#endif
```

Comment:
added local variables + fixed indent + corrected bracket position. ref: https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#formatting-conditions. 
```suggestion
		J9Class *declaringClass = J9_CLASS_FROM_METHOD(methodID->method);
		J9UTF8 *methodName = J9ROMMETHOD_NAME(romMethod);
		if (isPolymorphicMHMethod(currentThread->javaVM, declaringClass, methodName)
#if JAVA_SPEC_VERSION >= 9
			|| ((declaringClass == J9VMJAVALANGINVOKEVARHANDLE(currentThread->javaVM))
				&& VM_VMHelpers::isPolymorphicVarHandleMethod(J9UTF8_DATA(methodName), J9UTF8_LENGTH(methodName)))
#endif
```

Reply:
**Note:** RI tags methods with `@PolymorphicSignature` annotation during ClassFile parsing. So, they only have to do a bit test for identifying polymorphic methods in native code. On the other hand, OJ9 has to compare class and method names for identifying polymorphic methods since it does not have extra bits to tag methods. The above `MethodHandleNatives.init` string comparisons will be repeated from `MemberName`'s constructor: https://github.com/ibmruntimes/openj9-openjdk-jdk16/blob/a6d3f24ce5f8ec72bf3172f24833c2108ebc0dce/src/java.base/share/classes/java/lang/invoke/MemberName.java#L585-L608. OJ9 will do the `strcmps` twice, which can be improved in the future.

Reply:
If this is a perf concern, can we reclaim the `@FrameIteratorSkip` annotation and its associated bit for use by `@PolymorphicSignature` methods.  The `@FrameIteratorSkip` is only used by the OJ9 implementation of MH

Reply:
Opened https://github.com/eclipse-openj9/openj9/issues/12599 to track reclaiming of the `@FrameIteratorSkip` bit.